src/components/resizable.tsx

```diff
@@ -0,0 +1,164 @@
+import '@isrd-isi-edu/chaise/src/assets/scss/_resizable.scss';
+import { useRef, useEffect, useState } from 'react';
+import { fireCustomEvent } from '@isrd-isi-edu/chaise/src/utils/ui-utils';
+
+/* 
+Component Usage:
+    This component is used to implement a horizontal dragable and resizable component/div
+
+Props:
+    left - the left-pane component in the resizable layout
+    right - the right-pane component in the resizable layout
+    className - to apply custom styles on the component
+    minWidth - the minimum width limit on the left-pane component | defaults to 250px
+    maxWidth - the maximum width limit on the left-pane component | defaults to 450px
+    initialWidth - the initial width for the left-pane component | defaults to 270px
+    convertMaxWidth - use this flag in case you pass maxWidth in vw units instead of px units
+
+Note:
+    To resize other components based on the resized value left-pane component add an event listener and listen to the 
+    'resizable-width-change' event.
+
+    For usage refer to the recordset.tsx component
+*/
+
+
+const vwToPx = (value: number) => {
```

Comment:
Please add a jsdoc style comment to the function and explain what it does

```ts
/**
 * <description goes here>
 * @param {number} value <explain the value>
 */ 
const vwToPx = (value: number) => {
```

Reply:
@RFSH should I move this function into ui-utils ?

Reply:
Yeah that's a good idea. Make sure to name it something better. Maybe `convertViewWidthToPixel` or `convertVWToPixel`.

Reply:
done